airflow/providers/google/cloud/triggers/bigquery.py

```diff
@@ -118,7 +132,8 @@ async def run(self) -> AsyncIterator[TriggerEvent]:  # type: ignore[override]
                     await asyncio.sleep(self.poll_interval)
         except asyncio.CancelledError:
             self.log.info("Task was killed.")
-            if self.job_id and self.cancel_on_kill:
+            if self.job_id and self.cancel_on_kill and self.safe_to_cancel():
```

Comment:
```suggestion
            if self.job_id and self.cancel_on_kill and self.task_instance.state not in {
            TaskInstanceState.RUNNING,
            TaskInstanceState.DEFERRED,
        }:
```
I feel you don't need the extra function if you only want to check if the state is in the above states. safe_to_cancel gives the impression that there's a lot you're checking. 
Also, check if it should be `TaskInstanceState.unfinished`

Reply:
Is there `TaskInstanceState.unfinished`? I could not find one

Reply:
Also this method was to reduce code complexity as there was already a try-except with if else

Reply:
> Is there TaskInstanceState.unfinished? I could not find one

Sorry, I meant State.unfinished but that includes dagrun states

Reply:
I have changed the logic now as it requires to query fresh task instance state